app/com/linkedin/drelephant/mapreduce/MapReduceFSFetcherHadoop2.java

```diff
@@ -0,0 +1,305 @@
+  }
+
+  private DataFiles getHistoryFiles(AnalyticJob job) throws IOException {
+    String jobId = Utils.getJobIdFromApplicationId(job.getAppId());
+    String jobConfPath = null;
+    String jobHistPath = null;
+
+    // Search files in done dir
+    String jobHistoryDirPath = getHistoryDir(job);
+    RemoteIterator<LocatedFileStatus> it;
+    try {
+      it = _fs.listFiles(new Path(jobHistoryDirPath), false);
+    } catch (FileNotFoundException e) {
+      logger.error("Can't find job info of " + jobId + ": directory " + jobHistoryDirPath + " not found");
+      return null;
+    }
+    while (it.hasNext() && (jobConfPath == null || jobHistPath == null)) {
+      String name = it.next().getPath().getName();
+      if (name.contains(jobId)) {
+        if (name.endsWith("_conf.xml")) {
+          jobConfPath = jobHistoryDirPath + name;
+        } else if (name.endsWith(".jhist")) {
+          jobHistPath = jobHistoryDirPath + name;
+        }
+      }
+    }
+
+    // If some files is missing, search in the intermediate-done-dir in case that HistoryServer has not
+    // move them into the done-dir.
+    String intermediateDirPath = _intermediateHistoryLocation + File.separator + job.getUser() + File.separator;
+    if (jobConfPath == null) {
+      jobConfPath = intermediateDirPath + jobId + "_conf.xml";
+      if (!_fs.exists(new Path(jobConfPath))) {
+        throw new FileNotFoundException("Can't find config of " + jobId + " in neither "
+                + jobHistoryDirPath + " nor " + intermediateDirPath);
+      }
+      logger.info("Found job config in intermediate dir: " + jobConfPath);
+    }
+    if (jobHistPath == null) {
+      try {
+        it = _fs.listFiles(new Path(intermediateDirPath), false);
+        while (it.hasNext()) {
+          String name = it.next().getPath().getName();
+          if (name.contains(jobId) && name.endsWith(".jhist")) {
+            jobHistPath = intermediateDirPath + name;
+            logger.info("Found history file in intermediate dir: " + jobHistPath);
+            break;
+          }
+        }
+      } catch (FileNotFoundException e) {
+        logger.error("Intermediate history directory " + intermediateDirPath + " not found");
+      }
+      if (jobHistPath == null) {
+        throw new FileNotFoundException("Can't find history file of " + jobId + " in neither "
+                + jobHistoryDirPath + " nor " + intermediateDirPath);
+      }
+    }
+
+    return new DataFiles(jobConfPath, jobHistPath);
+  }
+
+  @Override
+  public MapReduceApplicationData fetchData(AnalyticJob job) throws IOException {
+    DataFiles files = getHistoryFiles(job);
+    String confFile = files.getJobConfPath();
+    String histFile = files.getJobHistPath();
+    String appId = job.getAppId();
+    String jobId = Utils.getJobIdFromApplicationId(appId);
+
+    JobHistoryParser parser = new JobHistoryParser(_fs, histFile);
```

Comment:
You should probably put this stuff in a try-catch block to match `MapReduceFetcherHadoop2.java`